src/resources/createResourceRoutes.js

```diff
@@ -47,100 +56,147 @@ export const createKubernetesUrl = ({
+}) => {
   const { isEnabled: isColumnLeyoutEnabled } = useFeature('COLUMN_LAYOUT');
-  const layoutState = useRecoilValue(columnLayoutState);
+  const [layoutState, setLayoutColumn] = useRecoilState(columnLayoutState);
+  const [searchParams] = useSearchParams();
+  const layout = searchParams.get('layout');
+  const { resourceListUrl } = useUrl();
 
-  const elementListProps = usePrepareListProps(props);
-  const elementDetailsProps = usePrepareDetailsProps({
-    ...props,
-    customResourceName: layoutState?.midColumn?.resourceName,
-    customNamespaceId: layoutState.midColumn?.namespaceId,
-  });
+  const { t } = useTranslation();
 
-  const listComponent = React.cloneElement(children, {
-    ...elementListProps,
-    enableColumnLayout:
-      elementListProps.resourceType !== 'Namespaces'
-        ? isColumnLeyoutEnabled
-        : false,
-  });
-  const detailsComponent = React.cloneElement(details, {
-    ...elementDetailsProps,
-  });
+  const {
+    resourceName: resourceNameFromParams,
+    namespaceId: namespaceIdFromParams,
+  } = useParams();
 
-  return (
-    <FlexibleColumnLayout
-      style={{ height: '100%' }}
-      layout={layoutState?.layout || 'OneColumn'}
-      startColumn={<div slot="">{listComponent}</div>}
-      midColumn={<div slot="">{detailsComponent}</div>}
-    />
+  const resourceName = useMemo(
+    () => props.resourceName ?? resourceNameFromParams,
+    [props.resourceName, resourceNameFromParams],
+  );
+  const namespaceId = useMemo(
+    () => props.namespaceId ?? namespaceIdFromParams,
+    [props.namespaceId, namespaceIdFromParams],
   );
-};
-
-const DetailsWrapper = ({ children, list, ...props }) => {
-  const { isEnabled: isColumnLeyoutEnabled } = useFeature('COLUMN_LAYOUT');
-  const [layoutState, setLayoutColumn] = useRecoilState(columnLayoutState);
-
-  const [searchParams] = useSearchParams();
-  const layout = searchParams.get('layout');
 
   const initialLayoutState = layout
     ? {
-        layout: isColumnLeyoutEnabled ? layout : 'OneColumn',
+        layout: isColumnLeyoutEnabled && layout ? layout : layoutState?.layout,
         midColumn: {
-          resourceName: props.resourceName,
+          resourceName: resourceName,
           resourceType: props.resourceType,
-          namespaceId: props.namespaceId,
+          namespaceId: namespaceId,
         },
         endColumn: null,
       }
     : null;
 
   useEffect(() => {
-    if (layout && props.resourceName && props.resourceType) {
+    if (layout && resourceName && props.resourceType) {
       setLayoutColumn(initialLayoutState);
     }
-  }, [layout, props.resourceName, props.resourceType, props.namespaceId]); // eslint-disable-line react-hooks/exhaustive-deps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    layout,
+    isColumnLeyoutEnabled,
+    namespaceId,
+    resourceName,
+    props.resourceType,
+  ]);
+
+  const layoutCloseCreateUrl = resourceListUrl({
+    kind: props.resourceType,
```

Comment:
Maybe that will be good to have "normal" prop `resourceType` and `resourceName` to not pass it hidden in `props`
```suggestion
    kind: resourceType,
```